### ISSUE TITLE
Autostart countdown timers via exported activity

### ISSUE THREAD

**Issue by danielschudel** [dpadgett/Timer#4]
I was thinking about your idea to autostart countdown timers. As far as I can tell, the app "NFC Task Launcher" only allows you to specify an applications activity to start. It does not allow any parameters to be passed. I don't know if this is a shortcoming of "NFC Task Launcher" or of Android.

Alternatives:

1 - Export multiple CountdownTimerAutostartX activities for standard lengths of time. For example, CountdownTimerAutostart5minutes, CountdownTimerAutostart15minuntes, CountdownTimerAutostart60minutes. When the activity is launched, and there isn't a currently running timer, then start one for the time specified. If there is a timer running, then add the length of time to the current running timer. The user could setup a 20 minute timer by using the 5 and 15 minute NFC tags.

2 - Export a single CountdownTimerAutostart that simply takes you straight to the countdown Timer tab.

3 - No reason you can't do both 1 and 2.

Thoughts?

**Comment by danielschudel** [comment]
4 - Export multiple CountdownTimerAutostartSlot[1-9] activities. In the Timer app, configure and remember the time duration associated with Slot[1-9]. This would allow the user to have the greatest control over the Timer duration, but is a bit more abstract dealing with "Slots".

**Comment by dpadgett** [comment]
yeah I was thinking of option 4 too... it's kind of clunky though, since it artificially limits the number of different timers you could do.  On the other hand it looks like there's not a way to dynamically add an activity.  Also there already seems to be a count-down target in the task launcher app, but I'm not sure how it works since I don't have any of the tags.

**Comment by danielschudel** [comment]
I tried out the two timer actions (hadn't paid attention to them before).

Task - "Set timer" - In Jelly Bean, you get a little icon in the notification area when the timer finishes. No sounds or popup alerts. Not too useful, but probably something the authors of "NFC Task Launcher" could improve on.

Task - "Set alarm (timer)" - This one does give an alert with your configured alarm sound when the timer expires. It doesn't have duration resolution below a minute, so, your alarms could expire up to 59 seconds early. It does allow multiple timers "in-flight" and each alerts accordingly. This matches closest to what you were proposing.

**Comment by danielschudel** [comment]
I picked up my tags from here: http://www.ebay.com/itm/261082337397?ssPageName=STRK:MEWNX:IT&_trksid=p3984.m1497.l2649

Do you have an NFC capable phone/tablet? If so, I don't mind snail mailing you one of my extras. Just drop me an email.

**Comment by dpadgett** [comment]
Yeah, I have a gnex and two nexus 7s.  I can order my own, just have to get
around to it.  It's definitely possible in the android api to pass in
extras when you launch an activity, just not with the nfc launcher I guess.
 Another option would be to have the activity start the countdown at
whatever time you left it set to.  But to be honest I'm kind of leaning
toward option 4, I'm just having trouble thinking of a way to put it into
the UI without making it cluttered.
